subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2545,72 +2560,69 @@ int lwm2m_write_handler(struct lwm2m_engine_obj_inst *obj_inst,
 			ret = lwm2m_write_handler_opaque(obj_inst, res,
 							 res_inst, msg,
 							 data_ptr, data_len);
-			if (ret < 0) {
-				return ret;
-			}
-
 			len = ret;
 			break;
 
 		case LWM2M_RES_TYPE_STRING:
-			engine_get_string(&msg->in, write_buf, write_buf_len);
+			ret = engine_get_string(&msg->in, write_buf,
+						write_buf_len);
 			len = strlen((char *)write_buf);
```

Comment:
Isn't this potentially bad? If `engine_get_string` produced an error, which we now can see by checking `ret`, we have no idea what `write_buf` actually contains right? Of course, this is only a read, but it could potentially read indefinitely, right, since we have no way of knowing that the string is terminated?